app/controllers/api/automated_deploys_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+# deploys that automatically get triggered when new hosts come up or are restarted
+class Api::AutomatedDeploysController < Api::BaseController
+  STAGE_NAME = 'Automated Deploys'
+
+  before_action :find_or_create_stage
+  before_action :find_deploy_group
+  before_action :find_last_deploy
+
+  def create
+    deploy_service = DeployService.new(current_user)
+    deploy = deploy_service.deploy!(
+      @stage,
+      reference: @last_deploy.reference,
+      buddy_id: @last_deploy.buddy_id,
```

Comment:
Or maybe we have a special "auto-deploy-buddy" user that can act as a stand-in for a real buddy.